kad-dht/README.md

```diff
@@ -490,7 +490,9 @@ These are the requirements for each `MessageType`:
 
 * `FIND_NODE`: In the request `key` must be set to the binary `PeerId` of the
   node to be found. In the response `closerPeers` is set to the `k` closest
-  `Peer`s.
+  `Peer`s.  Each closer peer will be a DHT server, unless `key` is an exact match
+  for a peer known to the node, in which case the closer peers will include the
+  matching peer, even if it is not a DHT server.
```

Comment:
Should other close peers be included even if the node knows the peer being searched for?

```suggestion
  for a peer known to the node, in which case the closer peers will only include the
  matching peer, even if it is not a DHT server.
```

Reply:
I would say that including the `k` closest peers in important, even for the `FIND_NODE` RPC when the target is known since `FIND_NODE` can also be used to find the closest peers to a given peer.

Eg. when refreshing the routing table, nodes look for the closest nodes to themselves. If the DHT servers only respond with the requester's address, it will be harder for the requester to discover its closest peers.